Update JOOQ reference docs (#83)
hasura/promptql-docs: docs/reference/connectors/_jdbcConfig.mdx

## Native queries {#native-queries}

Native queries allow you to use the SQL syntax of the underlying data source to create custom operations and expose them as models in your application. This is useful for complex queries, stored procedures, or custom functions that you want to leverage directly in your API.

### Native query structure

A native query is a single SQL statement that returns results and can take arguments. The JDBC connector supports two methods for defining native queries:

1. **File-based approach** (recommended): Store SQL queries in separate files
2. **Configuration-based approach**: Define queries directly in the `configuration.json` file

### File-based native queries

To create a file-based native query:

1. Create a directory structure for your native operations:
   ```bash
   mkdir -p <subgraph>/connector/<connector>/native_operations/queries/
   ```

2. Create a SQL file with your query, using `{{parameter}}` syntax for parameters:
   ```sql
   -- <subgraph>/connector/<connector>/native_operations/queries/get_customers_by_region.sql

>    SELECT * FROM customers
>    WHERE region = {{region}}
>    AND sales > {{min_sales}}

Reviewer: ndc-jdbc native queries use colon based prefix
```suggestion
   SELECT * FROM customers
   WHERE region = :region
   AND sales > :min_sales
```

   ```

4. Update your metadata to track the new native query:
   ```bash
   ddn connector-link update <connector_name> --add-all-resources
   ```

### Configuration-based native queries

You can also define native queries directly in the `configuration.json` file:

```json
{
  "version": "v2",
  "connection_uri": {
    "value": "..."
  },
  "schemas": ["public"],
  "tables": [...],
  "functions": [
    {
      "name": "get_customers_by_region",
      "description": "Get customers filtered by region",
      "sql": "SELECT * FROM customers WHERE region = :region AND sales > :min_sales",
      "parameters": [
        {
          "name": "region",
          "description": "Region to filter by",
          "type": {
            "scalar_type": "STRING"
          }
        },
        {
          "name": "min_sales",
          "description": "Minimum sales amount",
          "type": {
            "scalar_type": "INT64"
          }
        }
      ],
      "result_type": {
        "type": "array",
        "element_type": {
          "type": "named",
          "name": "public.customers"
        }
      }
    }
  ]
}
```

> ### Parameter syntax
> 
> The JDBC connector supports two parameter syntaxes:
> 
> 1. **Colon prefix syntax**: `:parameter` (used in configuration-based approach)
> 2. **Double curly braces syntax**: `{{parameter}}` (used in file-based approach)
> 
> Both syntaxes are translated to parameterized queries, which helps prevent SQL injection attacks.

Reviewer: ```suggestion
### Parameter syntax

The JDBC connector supports colon prefix syntax to specify parameters. 
This syntax is translated to parameterized queries, which helps prevent SQL injection.
```

### Important syntax rules

> When writing native queries, follow these rules:
> 
> 1. **Parameters as scalar values only**: Parameters can only be used in place of scalar values, not table names, column names, or other SQL parts
> 2. **No quoting of string parameters**: Don't add quotes around parameters (use `{{name}}` not `'{{name}}'`)
> 3. **Single statements only**: Each native query should be a single SQL statement without a semicolon at the end
> 4. **String patterns with concatenation**: For LIKE patterns, use concatenation (e.g., `LIKE '%' || {{search}} || '%'`)
> 5. **No "hasura_" prefixed parameters**: Parameter names starting with `hasura_` are reserved

Reviewer: ```suggestion
When writing native queries, follow these rules:

1. **Parameters as scalar values only**: Parameters can only be used in place of scalar values, not table names, column names, or other SQL parts
2. **No quoting of string parameters**: Don't add quotes around parameters (use `:name` not `':name'`)
3. **Single statements only**: Each native query should be a single SQL statement without a semicolon at the end
4. **String patterns with concatenation**: For LIKE patterns, use concatenation (e.g., `LIKE '%' || :search || '%'`)
5. **No "hasura_" prefixed parameters**: Parameter names starting with `hasura_` are reserved
```
